src/python/dxpy/workflow_builder.py

```diff
@@ -68,6 +68,33 @@ def _parse_executable_spec(src_dir, json_file_name, parser):
         except Exception as e:
             raise WorkflowBuilderException("Could not parse {} file as JSON: {}".format(json_file_name, e.args))
 
+def _fetch_spec_from_workflow(args, parser):
+    try: 
+        json_spec = dxpy.api.workflow_describe(args._from)
+        json_spec.update(version=args.version_override)
+        json_spec = _cleanup_empty_keys(json_spec)
+
+    except Exception as e:
+        raise WorkflowBuilderException("Could not get specs from given workflow {}: {}".format(args._from, e.args))
+
+    return json_spec
+
+def _cleanup_empty_keys(json_spec):
+    import re
+    clean_json = re.sub('"\w*": null,','',json.dumps(json_spec))
+    
+    return json.loads(clean_json)
+
+def _check_dxcompile_version(json_spec):
+    if "tags" in json_spec and "dxCompiler" in json_spec["tags"]:
+        if "details" in json_spec and "version" in json_spec["details"]:
+            import requests
+            from distutils.version import StrictVersion
+            url = 'https://github.com/dnanexus/dxCompiler/releases/latest'
```

Comment:
You don't want to always enforce it's the latest version of dxCompiler -- that is going to change over time. The version to compare against will be 1 specific version (the release of dxCompiler where all features on that side are implemented.) Maybe set a placeholder number here with a comment to update it later, like 100.100.100